ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+                handler_name = get(handler, 'name')
+                event_handler = {handler_name: {}}
+                action = get(handler, 'action')
+                action_type = get(handler, 'action_type')
+                delay = get(handler, 'delay')
+                trigger = get(handler, 'trigger')
+                regex = get(handler, 'regex')
+                if action is not None and action_type is not None:
+                    event_handler[handler_name].update({'action_type': action_type})
+                    event_handler[handler_name].update({'action': action})
+                if delay is not None:
+                    event_handler[handler_name].update({'delay': delay})
+                if get(handler, 'asynchronous') is True:
+                    event_handler[handler_name].update({'asynchronous': 'handler.asynchronous'})
+                if trigger is not None:
+                    event_handler[handler_name].update({'trigger': trigger})
+                if regex is not None:
+                    event_handler[handler_name].update({'regex': regex})
+            return event_handler
+        return None
+
+    @cached_property
+    def load_interval(self):
+        load_interval_default = get(self._hostvars, 'load_interval_default')
+        if load_interval_default is not None:
+            return {'default': load_interval_default}
+        return None
+
+    @cached_property
+    def queue_monitor_length(self):
+        queue_monitor_length = get(self._hostvars, 'queue_monitor_length')
+        if queue_monitor_length is not None:
+            queue_monitor_length_dict = {'enabled': True}
+            queue_monitor_length_notifying = get(queue_monitor_length, 'notifying')
+            feature_support = get(self._platform_settings, 'feature_support')
+            queue_monitor_length_notify = ""
+            if feature_support is not None:
+                queue_monitor_length_notify = get(feature_support, 'queue_monitor_length_notify')
+            if (
+                    queue_monitor_length_notifying is not None and
+                    queue_monitor_length_notify is not False
+            ):
+                queue_monitor_length_dict.update({'notifying': queue_monitor_length_notifying})
+            if get(queue_monitor_length, 'log') is not None:
+                queue_monitor_length_dict.update({'log': get(queue_monitor_length, 'log')})
+            return queue_monitor_length_dict
+        return None
+
+    @cached_property
+    def name_server(self):
+        name_servers = get(self._hostvars, 'name_servers')
+        if name_servers is not None:
+            return {
+                'source':
+                    {
+                        'vrf': self._mgmt_interface_vrf
+                    },
+                'nodes': name_servers
+            }
+        return None
+
+    @cached_property
+    def redundancy(self):
+        redundancy = get(self._hostvars, 'redundancy')
```

Comment:
refactor to get `redundancy.protocol` directly. You can just `return get(self._hostvars, 'redundancy.protocol')` since it will return `None` if not found.

Reply:
Same elsewhere